include/mata/nfa.hh

```diff
@@ -374,65 +381,49 @@ public:
         return post[q];
     };
 
-    void emplace_back() { post.emplace_back(); }
+    void emplace_back()
+    {
+        post.emplace_back();
+        if (post.size()-1 > max_state_)
+            ++max_state_;
+    }
+
+    void clear()
+    {
+        post.clear();
+        max_state_ = 0;
+    }
 
-    void clear() { post.clear(); }
     bool empty() const { return post.empty(); }
-    void resize(size_t n) { post.resize(n); }
-    size_t size() const { return post.size(); }
 
-    /**
-     * Function removes empty indices in transition vector and renames states accordingly
-     * @return Renaming of states where on index defined by old state number is a new number of the same state
-     */
-    std::vector<State> defragment()
+    void resize(size_t n)
     {
-        std::vector<State> renaming(this->post.size());
-        std::vector<State> removed{};
-
-        size_t last_empty = 0;
-        const size_t post_size = post.size();
-        for (size_t i = 0; i < post_size; ++i) {
-            if (post.at(i).empty()) {
-                last_empty = i;
-                removed.push_back(i);
-            } else if (last_empty < i) {
-                post[last_empty] = post[i];
-                renaming[i] = last_empty;
-                last_empty = i;
-            } else { // there was no empty space, last_empty is synchronized with i
-                renaming[i] = i; // nothing changed, no renaming needed
-                last_empty += 1;
-            }
-        }
+        post.resize(n);
+        if (post.size()-1 > max_state_)
+            max_state_ = post.size();
```

Comment:
I'm debuggin some issues in Python binding, and I think, there might be an error. Should this be max_state_ = post.size() - 1? Because, if we resize the post to e.g. 12, then the max state should be 11, right? @martinhruska can you check if my reasoning is correct? This would fix the issues in binding as well, I think.

Reply:
yes, you are probably right, it should be `max_state_ =  post.size()-1` because counting from zero.